server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -1540,6 +1551,48 @@ int insertPendingSegmentsIntoMetastore(
     return Arrays.stream(updated).sum();
   }
 
+  @VisibleForTesting
+  public void insertUsedSegments(Set<DataSegment> dataSegments, Map<SegmentId, String> upgradedFromSegmentIdMap)
```

Comment:
This method should not be in this class as it is invoked only from tests and does not update any state of this class.

Reply:
Given the structure of IngestionTestBase, it seems difficult to do this without duplication

Reply:
Could you please elaborate?

As a last resort, duplication in tests is fine but we shouldn't have to include unnecessary logic in classes as it makes the code confusing.

Reply:
Moved it to a test class as a static method

Reply:
That's great, thanks!